src/components/ec-table-head/ec-table-head.vue

```diff
@@ -49,47 +49,52 @@
   </thead>
 </template>
 
-<script setup>
-import VEcTooltip from '../../directives/ec-tooltip';
+<script setup lang="ts">
+import type { StyleValue } from 'vue';
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import vEcTooltip from '../../directives/ec-tooltip';
+import type { SortDirection } from '../../enums';
 import EcIcon from '../ec-icon';
+import { IconName, IconType } from '../ec-icon/types';
 import EcTableSort from '../ec-table-sort';
+import type {
+  StickyColumnPosition, TableHeadColumn, TableHeadEvent, TableHeadEvents, TableHeadSort,
+} from './types';
+
+const emit = defineEmits<{
+  'sort': [value: TableHeadEvents[TableHeadEvent.SORT]],
+}>();
+
+interface TableHeadProps {
+  columns?: TableHeadColumn[],
+  sorts?: TableHeadSort[],
+  stickyColumn?: StickyColumnPosition,
+}
 
-const emit = defineEmits(['sort']);
-
-const props = defineProps({
-  columns: {
-    type: Array,
-    default: () => [],
-  },
-  sorts: {
-    type: Array,
-    default: () => [],
-  },
-  stickyColumn: {
-    type: String,
-    validator(value) {
-      return ['left', 'right'].includes(value);
-    },
-  },
+const props = withDefaults(defineProps<TableHeadProps>(), {
+  columns: () => [],
+  sorts: () => [],
 });
 
-function getSortDirection(column) {
+function getSortDirection(column: TableHeadColumn): SortDirection | undefined {
   const existingSort = props.sorts.find(sort => sort.column === column.name);
   return existingSort && existingSort.direction;
 }
 
-function onSort(column) {
+function onSort(column: TableHeadColumn): void {
   emit('sort', column);
 }
 
-function getWidthStyle(column) {
+function getWidthStyle(column: TableHeadColumn): StyleValue {
   if (column && (column.maxWidth || column.minWidth)) {
     return { maxWidth: column.maxWidth, minWidth: column.minWidth };
   }
-  return null;
+  return null as unknown as StyleValue;
```

Comment:
```suggestion
  return null;
```
null doesn't match type StyleValue

Reply:
Where was the problem in template or in this function?

Reply:
In the template

Reply:
I think we could return undefined then, instead of null.